exonum/src/runtime/dispatcher/mod.rs

```diff
@@ -117,9 +117,20 @@ impl Dispatcher {
         spec: InstanceSpec,
         constructor: Vec<u8>,
     ) -> Result<(), ExecutionError> {
-        // Start the built-in service instance.
+        // Add the service instance to the service.
         ExecutionContext::new(self, fork, Caller::Blockchain)
-            .start_adding_service(spec, constructor)?;
+            .start_adding_service(spec.clone(), constructor)?;
+        // Mark service as active.
+        self.activate_pending(fork);
+        // Remove service from pending.
+        Schema::new(fork as &Fork).take_pending_instances();
+        // Flush changes, so the snapshot will be created with all the recently added changes.
+        // We can do it safely, since any failure on the genesis block level will result in the
+        // overall blockchain creation failure.
+        fork.flush();
+        // Start service within runtime.
+        self.start_service((fork as &Fork).snapshot(), &spec)
```

Comment:
This is very much a hack relying on the peculiarities of `impl RawAccess for Fork`. I'd much prefer if all services were activated at once and based on a `Patch` - like it was before, with the exception that `after_transactions` would be called after that.